fix: add port to SMTP host label

eliobischof opened the pull request:
Closes #4737 

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

livio-a commented on internal/command/smtp.go:
```diff
@@ -98,14 +98,17 @@ func (c *Commands) RemoveSMTPConfig(ctx context.Context) (*domain.ObjectDetails,
+		if hostAndPort = strings.TrimSpace(hostAndPort); hostAndPort == "" {
 			return nil, errors.ThrowInvalidArgument(nil, "INST-SF3g1", "Errors.Invalid.Argument")
 		}
+		if strings.Count(hostAndPort, ":") != 1 {
+			return nil, errors.ThrowInvalidArgument(nil, "INST-VDwvq", "Errors.Invalid.Argument")
+		}
```
please use an appropriate function like `net.SplitHostPort`, which will also correctly handle ipv6 addresses, ...

livio-a commented on internal/command/smtp.go:
```diff
@@ -143,12 +146,15 @@ func (c *Commands) prepareAddSMTPConfig(a *instance.Aggregate, from, name, host,
+		if hostAndPort = strings.TrimSpace(hostAndPort); hostAndPort == "" {
+			return nil, errors.ThrowInvalidArgument(nil, "INST-VDwvq", "Errors.Invalid.Argument")
+		}
+		if strings.Count(hostAndPort, ":") != 1 {
```
please see above

livio-a commented on internal/command/smtp_test.go:
```diff
@@ -184,6 +184,28 @@ func TestCommandSide_AddSMTPConfig(t *testing.T) {
+			res: res{
+				err: caos_errs.IsErrorInvalidArgument,
+			},
+		},
```
maybe we could add a ipv6 test as well

eliobischof commented:
Good catch @livio-a, IPv6 is valid and tested now.
Did you also review the console changes, or should @peintnermax review that part?

peintnermax commented:
> Good catch @livio-a, IPv6 is valid and tested now. Did you also review the console changes, or should @peintnermax review that part?

Console changes look good to me 👍

eliobischof commented:
@peintnermax, FYI I just added the type=password attribute to the password component [7d4f09d](https://github.com/zitadel/zitadel/pull/4980/commits/7d4f09d4516b65063a1450dcc0efe084d0fee5f6)

livio-a reviewed (approved):
backend is now ok for me

peintnermax commented:
> @peintnermax, FYI I just added the type=password attribute to the password component [7d4f09d](https://github.com/zitadel/zitadel/pull/4980/commits/7d4f09d4516b65063a1450dcc0efe084d0fee5f6)

and autocomplete is not an issue or do we have to test this 🤔

eliobischof commented:
> and autocomplete is not an issue or do we have to test this thinking otherwise I would add a css class to show password field styling `-webkit-text-security: disc`

Good point! Can we live with this?

![image](https://user-images.githubusercontent.com/12727842/210794590-7bcb68b4-0c35-46df-beaa-430f3a456d46.png)

It's with type="password" and autocomplete="new-password".  I think it's the least bad solution: https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion#preventing_autofilling_with_autocompletenew-password.

I think it's better than showing the password and it's also better than `-webkit-text-security: disc`: https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-text-security

eliobischof commented:
I added the input attribute as decribed above https://github.com/zitadel/zitadel/pull/4980/commits/ac2200cf7df819f4420fed1764fb6a867b353171
I also fixed that the host field wasn't filled from remote data anymore https://github.com/zitadel/zitadel/pull/4980/commits/221212613c2c0d96905cf2d3517d3f95ee8eaff8

eliobischof commented:
@peintnermax, what do you think?

peintnermax commented:
IMO we should remove the type="password" and auto-complete attr as this could also prefill the field with the user password

peintnermax commented:
ok we should give this a try.. with new-password this should be fine 👍
